mg/cmd_internal_test.go

```diff
@@ -0,0 +1,75 @@
+package mg
+
+import (
+	"testing"
+)
+
+type nonSpec struct{ ActionType }
+
+func TestCmdSupport_Reduce_noCalls(t *testing.T) {
+	cs := &cmdSupport{}
+	ag, _ := NewAgent(AgentConfig{})
+	ctx, done := ag.NewCtx(nil)
+	defer close(done)
+
+	ctx.Action = nil
+	if state := cs.Reduce(ctx); state != ag.Store.State() {
```

Comment:
In general, you should compare with the State object you gave the reducer i.e. `!= ctx.State`.

Also, if this is checking that a reducer doesn't change the state when the action is not its primary action, it's not a safe assumption to make. The action is a hint to the reducer that it should do something, but they change the state by e.g. adding a status message based on things it did when the relevant action was dispatched.

Reply:
I think this is important to document its behaviour.